doc/stdenv/stdenv.chapter.md

```diff
@@ -116,6 +116,65 @@ On Linux, `stdenv` also includes the `patchelf` utility.
 
 ## Specifying dependencies {#ssec-stdenv-dependencies}
 
+Build systems often require more dependencies than just what `stdenv` provides. This section describes attributes accepted by `stdenv.mkDerivation` that can be used to make these dependencies available to the build system.
+
+### Simplified overview {#ssec-stdenv-dependencies-overview}
+A full reference of the many kinds of dependencies needed to cover all possible cases is provided in the next section, but here is a simplified overview of the most common kinds. It should cover most derivations.
+
+Add dependencies to `nativeBuildInputs` if they are executed during the build:
+- those which are needed on `$PATH` during the build, for example `cmake` and `pkg-config`
+- [setup hooks](#ssec-setup-hooks), for example [`makeWrapper`](#fun-makeWrapper)
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for build scripts (with the `--build` flag), which can be the case for e.g. `perl`
+
+Add dependencies to `buildInputs` if they will end up copied or linked into the final output or otherwise used at runtime:
+- libraries used by compilers, for example `zlib`,
+- interpreters needed by [`patchShebangs`](#patch-shebangs.sh) for scripts which are installed, which can be the case for e.g. `perl`
+
+Note that these criteria are not incompatible: for example software using Wayland usually needs the `wayland` library at runtime (so `wayland` should be added to `buildInputs`) but also executes the `wayland-scanner` program as part of the build to generate code (so `wayland` should also be added to `nativeBuildInputs`).
+
+Dependencies needed only to run tests are similarly classified between native (executed during build) and non-native (executed at runtime):
+- `nativeCheckInputs` for test tools needed on `$PATH` (like `ctest`) and [setup hooks](#ssec-setup-hooks) (for example [`pytestCheckHook`](#python))
+- `checkInputs` for libraries linked into test executables (for example the `qcheck` OCaml package)
+
+These dependencies are only injected when `doCheck` is set to `true`.
+
+Consider for example this simplified derivation for `solo5`, a sandboxing tool:
```

Comment:
```suggestion
#### Example

Consider this simplified derivation for `solo5`, a sandboxing tool:
```

Since the example is huge and the text before is quite long, it's worth having a subsection we can link to.